src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/GlobalJsonRemover.cs

```diff
@@ -27,13 +73,23 @@ public int OnAfterOpenSolution(object pUnkReserved, int fNewSolution)
             try
             {
                 Verify.HResult(solution.GetSolutionInfo(out string directory, out string solutionFile, out string optsFile));
-                ProjectItem globalJson = dte.Solution.FindProjectItem(Path.Combine(directory, "global.json"));
-                globalJson?.Remove();
+                var globalJsonPath = Path.Combine(directory, "global.json");
+                ProjectItem globalJson = dte.Solution.FindProjectItem(globalJsonPath);
+                globalJson?.Delete();
+                try
+                {
+
+                    _fileSystem.RemoveFile(globalJsonPath);
+                }
+                catch (FileNotFoundException) { }
```

Comment:
`RemoveFile` already does: https://github.com/dotnet/roslyn-project-system/blob/3f6c9b9995dde7adffa2044b6a5f97403d6852c7/src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/Win32FileSystem.cs#L30. However, checking for the existence of a file and then deleting it isn't a synchronous operation, nothing stops some other app deleting it between those calls.